src/Subject.ts

```diff
@@ -123,6 +123,12 @@ export class Subject<T> extends Observable<T> implements Observer<T>, Subscripti
     this._complete();
   }
 
+  asObservable(): Observable<T> {
+    return new Observable((observer: Observer<any>) => {
```

Comment:
Then make a derived class `SubjectObservable` or something, and it can do:

```TypeScript
export class SubjectObservable<T> extends Observable<T> {
  constructor(source: Subject<T>) {
    super();
    this.source = source;
  }
}
```

easy peasy.